File: app grifo att/lib/models/item_estoque.dart
class ItemEstoque {
  final int id;
  final String nome;
  final int quantidadeAtual;
  final String? categoria;
  final String? descricao;

  ItemEstoque({
    required this.id,
    required this.nome,
    required this.quantidadeAtual,
    this.categoria,
    this.descricao,
  });

  factory ItemEstoque.fromJson(Map<String, dynamic> json) {
    return ItemEstoque(
      id: json['id'] ?? 0,
      nome: json['nome'] ?? '',
      quantidadeAtual: json['quantidadeAtual'] ?? 0,
      categoria: json['categoria'],
      descricao: json['descricao'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'nome': nome,
      'quantidadeAtual': quantidadeAtual,
      'categoria': categoria,
      'descricao': descricao,
    };
  }
}

File: lab-prog-web/src/main/java/com/example/labprog/onepage/entity/ItemEstoque.java
package com.example.labprog.onepage.entity;

import jakarta.persistence.*;

@Entity
@Table(name = "itens_estoque")
public class ItemEstoque {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String nome;

    @Column(nullable = false)
    private Integer quantidadeAtual = 0;

    private String categoria;
    private String descricao;

    public ItemEstoque() {}

    public ItemEstoque(String nome, Integer quantidadeAtual, String categoria) {
        this.nome = nome;
        this.quantidadeAtual = quantidadeAtual;
        this.categoria = categoria;
    }

    // Getters e Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getNome() { return nome; }
    public void setNome(String nome) { this.nome = nome; }

    public Integer getQuantidadeAtual() { return quantidadeAtual; }
    public void setQuantidadeAtual(Integer quantidadeAtual) { this.quantidadeAtual = quantidadeAtual; }

    public String getCategoria() { return categoria; }
    public void setCategoria(String categoria) { this.categoria = categoria; }

    public String getDescricao() { return descricao; }
    public void setDescricao(String descricao) { this.descricao = descricao; }
}

